jwst/associations/lib/rules_level2b.py

```diff
@@ -252,7 +252,7 @@ def __init__(self, *args, **kwargs):
                     DMSAttrConstraint(
                         name='patttype',
                         sources=['patttype'],
-                        value=['2-point-nod|4-point-nod|along-slit-nod'],
+                        value=['2-point-nod|4-point-nod'],
```

Comment:
So I'm trying to wrap my head around the effect of this double negative: `along-slit-nod` used to be included in a `notany` constraint, which means we were *not* creating `spec2` ASN's for exposures with `along-slit-nod`? And now that that pattern type has been removed from the `notany` constraint, we *are* creating `spec2` ASN's for them?

But if that's the case, why aren't we creating `spec2` ASN's for dithered exposures using `2-point-nod` and `4-point-nod`? I'm not sure why those pattern types are treating differently, since they all should allow for potential image-from-image background subtraction, if the target is a point source.

Reply:
First: Correct, this "notany" means that none of the contained constraints can match. So, if patttype had any of the given values, the match would fail.

The removal of along-slit-nod allows this association to create the dedicated background associations for along-slit-nod, if and when they exist, as per the request of the issue.

The other patttypes, which are NIRSpec-specific, are handled by [Asn_Lv2NRSIFUNod](https://github.com/spacetelescope/jwst/blob/93b8d5dacbf15fb8b74c482b32df387e8d741042/jwst/associations/lib/rules_level2b.py#L783). Hence they are not allowed in this association.